analytics/tracker.go

```diff
@@ -0,0 +1,135 @@
+package analytics
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/bitrise-io/bitrise/configs"
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/bitrise/version"
+	"github.com/bitrise-io/go-utils/pointers"
+	"github.com/bitrise-io/go-utils/v2/analytics"
+	"github.com/bitrise-io/go-utils/v2/env"
+	stepmanModels "github.com/bitrise-io/stepman/models"
+	log "github.com/sirupsen/logrus"
+)
+
+// Tracker ...
+type Tracker interface {
+	SendWorkflowStarted(properties analytics.Properties, title string)
+	SendWorkflowFinished(properties analytics.Properties, failed bool)
+	SendStepStartedEvent(properties analytics.Properties, infoModel stepmanModels.StepInfoModel, inputs map[string]interface{})
+	SendStepFinishedEvent(properties analytics.Properties, results models.StepRunResultsModel)
+	Wait()
+}
+
+type tracker struct {
+	tracker       analytics.Tracker
+	envRepository env.Repository
+}
+
+// NewTracker ...
+func NewTracker() Tracker {
+	if isAnalyticsDisabled() {
+		return NoOpTracker{}
+	}
+	return tracker{
+		tracker:       analytics.NewDefaultTracker(),
+		envRepository: env.NewRepository(),
+	}
+}
+
+// SendWorkflowStarted ...
+func (t tracker) SendWorkflowStarted(properties analytics.Properties, title string) {
+	isCI := t.envRepository.Get(configs.CIModeEnvKey) == "true"
+	isPR := t.envRepository.Get(configs.PRModeEnvKey) == "true"
+	isDebug := t.envRepository.Get(configs.DebugModeEnvKey) == "true"
+	isSecretFiltering := t.envRepository.Get(configs.IsSecretFilteringKey) == "true"
+	isSecretEnvsFiltering := t.envRepository.Get(configs.IsSecretEnvsFilteringKey) == "true"
+	stateProperties := analytics.Properties{
+		"workflow_name":         title,
+		"ci_mode":               isCI,
+		"pr_mode":               isPR,
+		"debug_mode":            isDebug,
+		"secret_filtering":      isSecretFiltering,
+		"secret_envs_filtering": isSecretEnvsFiltering,
+	}
+	stateProperties.AppendIfNotEmpty("build_slug", t.envRepository.Get("BITRISE_BUILD_SLUG"))
```

Comment:
Isn't `isCI` redundant if we also send the build slug?

Reply:
I we can confidently say `isCI` iff `buildSlug!=nil` then it is redundant yeah. I'd honestly just keep it for the sake of consistency with other flags.

Reply:
Okay, it's fine